starter-code/src/App.js

```diff
@@ -1,28 +1,35 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
 import './App.css';
+import CountryDetail from './components/CountryDetail';
+import Navbar from './components/Navbar';
+import Sidebar from './components/Sidebar';
+import countries from './countries.json';
+import { Route } from 'react-router-dom';
 
 class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <p>
-            Edit <code>src/App.js</code> and save to reload.
-          </p>
-          <a
-            className="App-link"
-            href="https://reactjs.org"
-            target="_blank"
-            rel="noopener noreferrer"
-          >
-            Learn React
-          </a>
-        </header>
-      </div>
-    );
-  }
+	constructor() {
+		super();
+		this.state = {
+			countries: countries
+		};
+	}
+
+	render() {
+		return (
+			<div className="App">
+				<Navbar />
+				<Sidebar countries={this.state.countries} />
+
+				{this.state.countries.map((country, index) => {
+					let countryCCA3 = country.cca3;
+					console.log(countryCCA3);
+					return (
+						<Route key={index} path="/:countryCCA3" render={() => <CountryDetail country={country} />} />
```

Comment:
You can put your route in a switch element and define the path.  There is no need to map over the countries.  The Route path `/:countryCCA3` will take a parameter and render the correct `CountryDetail` component based on the params that you pass to it in your `Link` on the `Sidebar`.  Here in the `app.js` you can also define a function that returns the correct country based on the `cca3` value and pass that as props to your `CountryDetail` component.